### ISSUE TITLE
"sqlfluff fix --rules L036" corrupts "SELECT DISTINCT ID" query

### ISSUE THREAD

**Issue by barrywhart** [sqlfluff/sqlfluff#2765]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

`sqlfluff fix --rules L036` changes this query:
```
SELECT
DISTINCT id
```

to this:
```
SELECT DISTINCT id
DISTINCT
```

### Expected Behaviour

Did not expect the second `DISTINCT`, which now parses as an alias.

### Observed Behaviour

See above.

### How to reproduce

See above.

### Dialect

ANSI

### Version

Basically the latest `main`, commit ID `c19b634d0e87265667910b7c4a50c09ef780a2bb`

### Configuration

n/a

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by davesgonechina** [comment]
I reported this in Slack initially using essentially the [Default Configuration](https://docs.sqlfluff.com/en/stable/configuration.html#default-configuration) but using the `jinja` templater and `Snowflake` dialect, and the following excluded rules: L011,L029,L034,L031,L035,L042,L043,L044,L045

The issue occurs when the source is
```
SELECT 
        DISTINCT user_id 
    FROM {{ ref('base_prod__users') }}
```

which becomes
```
    SELECT user_id AS distinct
    FROM {{ ref('base_prod__users') }}
```

**Comment by barrywhart** [comment]
@tunetheweb: Oops, sorry. The test query I initially provided was missing a newline. Can you try again? Also, to be safe, make sure you are only running rule `L036`.

**Comment by tunetheweb** [comment]
Still not seeing it:

```
% more test.sql
SELECT DISTINCT id
% sqlfluff fix -f --rules L036 test.sql
==== finding fixable violations ====
==== no fixable linting violations found ====                                                                                                         
All Finished 📜 🎉!
% more test.sql
SELECT DISTINCT id
```

**Comment by barrywhart** [comment]
@davesgonechina: Can you create a new issue for this? The issue I was looking at is a bit different, and I can't reproduce you issue, so it would be good to get more detail.

**Comment by barrywhart** [comment]
@tunetheweb: Check the updated test query in my description. You need a newline between `SELECT` and `DISTINCT`.

**Comment by barrywhart** [comment]
Here's me reproducing it:
```
(sqlfluff-3.10.1) ➜  sqlfluff git:(main) ✗ cat test.sql                         
SELECT
DISTINCT id
(sqlfluff-3.10.1) ➜  sqlfluff git:(main) ✗ sqlfluff fix -f --rules L036 test.sql
==== finding fixable violations ====
== [test.sql] FAIL                                                                                                                                                                                      
L:   1 | P:   1 | L036 | Select targets should be on a new line unless there is                                                                                                                         
                       | only one select target.
==== fixing violations ====
1 fixable linting violations found
FORCE MODE: Attempting fixes...
Persisting Changes...
== [test.sql] PASS
Done. Please check your files to confirm.
(sqlfluff-3.10.1) ➜  sqlfluff git:(main) ✗ cat test.sql                         
SELECT DISTINCT id
DISTINCT
```

Notes that this does not occur in 0.10.1. It's a new issue since then (seen in `main`), most likely introduced by the changes I made when dealing with the post-fix parse issues.

**Comment by tunetheweb** [comment]
OK see it now.

**Comment by barrywhart** [comment]
@davesgonechina: As noted above, please create a new issue for this. Below are details on how I tried (but failed) to reproduce your issue.

I ran this command:
```
sqlfluff fix test.sql --dialect snowflake --exclude-rules L011,L029,L034,L031,L035,L042,L043,L044,L045
```

On this SQL file:
```
SELECT
        DISTINCT user_id
    FROM {{ ref('base_prod__users') }}
```


And this was the result:
```
SELECT DISTINCT user_id

FROM {{ ref('base_prod__users') }}
```

The blank line looks odd, but I don't see the `user_id AS distinct` output you mentioned.

**Comment by davesgonechina** [comment]
Done: https://github.com/sqlfluff/sqlfluff/issues/2769
